### PR TITLE
fix(toolbar): forbid redirecting to url not in allowlist

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

See issue 4 [here](https://posthog.slack.com/archives/C05276ME9T5/p1683572781345189).

## Changes

This PR:
1. forbids redirecting to urls not in the allowlist and 
2. uses a stricter regex for the wildcards in the allowed url list, effectively forbidding wildcards to match sub-sub-domains

## How did you test this code?

Added tests and checked for common app_urls and recording_domains in metabase.

### CODE REVIEW

File: posthog/api/decide.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def hostname_in_allowed_url_list(allowed_url_list: Optional[List[str]], hostname
 
     for permitted_domain in permitted_domains:
         if "*" in permitted_domain:
-            pattern = "^{}$".format(re.escape(permitted_domain).replace("\\*", "(.*)"))
+            pattern = "^{}$".format(re.escape(permitted_domain).replace("\\*", r"(\w*)"))
```
Comment (Twixes) [inline posthog/api/decide.py:45]: I think this breaks support for sub-subdomains like `app.dev.example.com`, do we have a test for that?

Comment (thmsobrmlr) [reply posthog/api/decide.py:45]: Yep, this forbids sub-subdomains intentionally as I found it surprising https://github.com/PostHog/posthog/pull/15557/files#diff-cbd48866da8bccafd5dc1f27e3b4cdb0327842dad31c3dfa3cdbd9e86afab0eaR772. Seems like it's an actual use case though? In that case I'm happy to revert this change.

**Comment by Twixes** [comment]
❗ Some red tests

**Comment by Twixes** [comment]
Something's still off 🤔

**Comment by thmsobrmlr** [comment]
> Something's still off 🤔

Yeah, I missed some old tests. Haven't looked into it now, since they relate to the apiUrl and we might remove that completely anyway..

**Comment by Twixes** [comment]
Bump!

**Comment by thmsobrmlr** [comment]
@Twixes Fixed the tests.

**Review by Twixes — APPROVED** [review]

**Pull request merged**